lib/validators/specResolver.js

```diff
@@ -268,7 +281,7 @@ class SpecResolver {
                 unresolvedDefinitions.push(() => {
                   if (result.definitions[defName].allOf) {
                     let matchFound = result.definitions[defName].allOf.some((item) => {
-                      return (item['$ref'] === `#${slicedLocalReferenceValue}` && !self.visitedEntities[`/definitions/${defName}`]);
+                      return (!self.visitedEntities[`/definitions/${defName}`]);
```

Comment:
why did we remove this? - Add comment like: Did not make sense of checking `slicedLocalReferenceValue` with the current `$ref` as they may or may not be same.